multitenant: unskip multitenant_admin_function_test

ecwall opened the pull request:
Fixes #95201

This test was flaking because it was verifying exact range start keys
which are non-deterministic.

This was fixed as part of #98483 which allows any value for the range start key
and can now be unskipped.

Release note: None

ecwall committed:
multitenant: unskip multitenant_admin_function_test

Fixes #95201

This test was flaking because it was verifying exact range start keys
which are non-deterministic.

This was fixed as part of #98483 which allows any value for the range start key
and can now be unskipped.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/98801)
<!-- Reviewable:end -->

ecwall committed:
multitenant: retry flaky multitenant_admin_function_test queries

Fixes #95252

Race conditions between the test and kv can cause the test to flake so retry
the query.

Release note: None

ecwall commented:
bors r=knz
